HTTP_API.md

```diff
@@ -1368,6 +1376,222 @@ The handler-specific descriptions below describe how each handler populates the
+    rule definition. This is informational and for display only.
+
+    `"archivalPeriodSeconds"`: a positive integer value that defines how long
+    Cryostat should wait, in seconds, between archiving snapshots of the
+    recording. The default setting is 30.
+
+    `"preservedArchives"`: a positive integer value that defines how many
+    archived copies of the recording should be kept in storage. When the number
+    of archived copies exceeds this number the oldest copies are deleted from
+    storage. The default setting is 1.
+
+    `"maxAgeSeconds"`: a positive integer value that defines the maximum age of
+    data to be retained in the active, in-memory Flight Recording within the
+    Target JVM. This can be used in combination with `"archivalPeriodSeconds"`
+    to minimize or eliminate data overlap between the end of one archived
+    recording and the start of the subsequent archived recording. If not
+    specified, the default setting is equal to `"archivalPeriodSeconds"`.
+
+    `"maxSizeBytes"`: a positive integer value that defines the maximum size, in
+    bytes, of the active in-memory Flight Recording within the Target JVM. If
+    the recording exceeds this memory size then event data will be dropped from
+    the recording. The default setting is unlimited.
+
+    ##### response
+    `201` - The result is the name of the created rule. The `LOCATION` header
+    will be set and its value will be the relative path to the created resource.
+
+    `400` - The rule definition could not be processed, either because the
+    provided document was malformed or invalid.
+
+    `401` - User authentication failed. The reason is an error message.
+    There will be an `X-WWW-Authenticate: $SCHEME` header that indicates
+    the authentication scheme that is used.
+
+    `409` - A rule with the same name already exists.
+
+    `415` - The request's `Content-Type` was invalid or unrecognized.
+
+    `500` - There was an unexpected error.
+
+    ##### example
+    ```
+    $ curl -X POST -F name="Test Rule" -F description="This is a rule for testing" -F targetAlias="io.cryostat.Cryostat" -F eventSpecifier="template=Continuous,type=TARGET" http://0.0.0.0:8181/api/v2/rules
+    < HTTP/1.1 201 Created
+    < location: /api/v2/rules/Test_Rule
+    < content-length: 79
+    {"meta":{"type":"text/plain","status":"Created"},"data":{"result":"Test_Rule"}}
+    ```
+
+* #### `RuleDeleteHandler`
+
+    ##### synopsis
+    Deletes a rule definition.
+
+    ##### request
+    `DELETE /api/v2/rules/:name`
+
+    ##### response
+    `200` - The result is empty. The rule was successfully deleted.
+
+    `401` - User authentication failed. The reason is an error message.
+    There will be an `X-WWW-Authenticate: $SCHEME` header that indicates
+    the authentication scheme that is used.
+
+    `404` - No rule with the given name exists.
+
+    `500`- An unexpected IOException occurred while deleting the rule
+    definition.
+
```

Comment:
For consistency, the `RulesPostHandler`, `TargetCredentialsPostHandler`, and `TargetCredentialsDeleteHandler` should probably have the same explanation for an HTTP 500 response since (from what I can tell) all four handlers give such a response only upon catching an `IOException`. Unless, unlike this handler, these 3 handlers can have other unexpected errors that aren't explicitly stated.

Reply:
> Unless, unlike this handler, these 3 handlers can have other unexpected errors that aren't explicitly stated.

That's generally the case for any `500` error - those are what you get if there is some unexpected internal error. I'm not really happy with the fact that other parts of the document list any specific reasons for `500`s, actually, because almost by definition this list is non-exhaustive, so including a partial list of reasons could be misleading for readers.

> all four handlers give such a response only upon catching an IOException

That's from a current reading of the explicit handling of an IOException, yes. But if the implementation changes at all and say a new NullPointerException becomes possible, that would also result in a 500 but would almost certainly not be explicitly handled and turned into a 500 by the handler - it would fall through to the WebServer failureHandler.